springwolf-plugins/springwolf-kafka-plugin/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/KafkaPayloadTypeExtractor.java

```diff
@@ -0,0 +1,78 @@
+package io.github.stavshamir.springwolf.asyncapi.scanners.channels;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.messaging.handler.annotation.Payload;
+import sun.reflect.generics.reflectiveObjects.ParameterizedTypeImpl;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.List;
+
+@Slf4j
+public class KafkaPayloadTypeExtractor {
+    public KafkaPayloadTypeExtractor() {
+    }
+
+    protected static Class<?> getPayloadType(Method method) {
+        String methodName = String.format("%s::%s", method.getDeclaringClass().getSimpleName(), method.getName());
+        log.debug("Finding payload type for {}", methodName);
+
+        Class<?>[] parameterTypes = method.getParameterTypes();
+        int parameterPayloadIndex = getPayloadParameterIndex(parameterTypes, method.getParameterAnnotations(), methodName);
+
+        return getPayloadParameterClass(method, parameterTypes, parameterPayloadIndex);
+    }
+
+    static Class<?> getPayloadParameterClass(Method method, Class<?>[] parameterTypes, int parameterPayloadIndex) {
+        Class<?> parameterClass = parameterTypes[parameterPayloadIndex];
+
+        try {
+            // Resolve generic type for kafka batch listeners
+            if (parameterClass == List.class) {
+                Type type = ((ParameterizedTypeImpl) method.getGenericParameterTypes()[parameterPayloadIndex]).getActualTypeArguments()[0];
+                parameterClass = Class.forName(type.getTypeName());
+            }
+        } catch (ClassNotFoundException ex) {
```

Comment:
I think the catch should be for `Exception`, it looks like `getActualTypeArguments` can also throw exceptions and may return an empty array. See: https://docs.oracle.com/javase/7/docs/api/java/lang/reflect/ParameterizedType.html

Reply:
Done